client/internal/httpsender_test.go

```diff
@@ -208,3 +208,127 @@ func TestRequestInstanceUidFlagReset(t *testing.T) {
 	assert.Equal(t, sender.receiveProcessor.clientSyncedState.flags, protobufs.AgentToServerFlags_AgentToServerFlags_Unspecified)
 	cancel()
 }
+
+func TestPackageUpdatesInParallel(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	localPackageState := NewInMemPackagesStore()
+	sender := NewHTTPSender(&sharedinternal.NopLogger{})
+	ch := make(chan struct{})
```

Comment:
```suggestion
	blockSyncCh := make(chan struct{})
```
To make it clear that this channel's purpose is to block the sync operation.